docs/proposal/20210311-single-controller-multitenancy.md

```diff
@@ -0,0 +1,321 @@
+- The `IdentityRef` on a `VsphereCluster` will be mutable. Setting the `IdentityRef` to nil will cause the controller to fallback to the static credentials
+- The controller will fail to delete a `VsphereClusterAccount` with an error if the account is used by `VsphereClusters`
+
+#### Clusterctl Changes
+
+Today, clusterctl move operates by tracking objectreferences within the same namespace, since we are now proposing to use cluster-scoped resources, we will need to add requisite support to clusterctl's object graph to track cluster-scoped resources that are used by the source cluster, and ensure they are moved. We will naively not delete cluster-scoped resources during a move, as they maybe referenced across namespaces. If a cluster uses a `Secret` for account credentials, the OwnerReference will get set by the controller and the `Secret` will be moved to the target cluster. Any `VsphereClusterAccount` that's required by the resources being moved, will need to be manually created on the target cluster prior to the move operation.
+
+### Security Model
+
+The intended RBAC model mirrors that for Service APIs:
+
+#### Roles
+
+For the purposes of this security model, 3 common roles have been identified:
+
+- **Infrastructure Provider**: The infrastructure provider (infra) is responsible for the overall environment that
+  the cluster(s) are operating in or the PaaS provider in a company.
+
+- **Management Cluster Operator**: The cluster operator (ops) is responsible for
+  administration of the Cluster API management cluster. They manage policies, network access,
+  application permissions.
+
+- **Workload Cluster Operator**: The workload cluster operator (dev) is responsible for
+  management of the cluster relevant to their particular applications .
+
+There are two primary components to the Service APIs security model: RBAC and namespace restrictions.
+
+#### RBAC
+
+RBAC (role-based access control) is the standard used for Kubernetes
+authorization. This allows users to configure who can perform actions on
+resources in specific scopes. RBAC can be used to enable each of the roles
+defined above. In most cases, it will be desirable to have all resources be
+readable by most roles, so instead we'll focus on write access for this model.
+
+##### Write Permissions
+
+|                              | VsphereClusterAccount | Secret | Cluster |
+| ---------------------------- | --------------------- | ------ | ------- |
+| Infrastructure Provider      | Yes                   | Yes    | Yes     |
+| Management Cluster Operators | Yes                   | Yes    | Yes     |
+| Workload Cluster Operator    | No                    | Yes    | Yes     |
+
+#### Namespace Restrictions
+
+- To prevent workload cluster operators from using cluster-scoped `VsphereClusterAccounts` that they should not be using, the `allowedNamespaces` will be used to dictate which namespaces are allowed to use the `VsphereClusterAccount`.
```

Comment:
```suggestion
- To prevent workload cluster operators from using cluster-scoped `VSphereClusterAccounts` that they should not be using, the `allowedNamespaces` will be used to dictate which namespaces are allowed to use the `VSphereClusterAccount`.
```